### PR TITLE
signals-backend: fan-out signal events in order to support scaled deployments

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

By using a unique subscriber ID for each instance all events will be fanned out to all instances and reach users regardless of what instance they're connected to.

👀  @drodil 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/signals-backend/src/service/SignalManager.ts [hunk 1]
```diff
@@ -62,10 +63,18 @@ export class SignalManager {
   }
 
   private constructor(options: SignalManagerOptions) {
+    this.events = options.events;
```
Comment (freben) [inline plugins/signals-backend/src/service/SignalManager.ts:66]: ...how did this ever work

**Review by freben — APPROVED** [review]

File: plugins/signals-backend/src/service/SignalManager.ts [hunk 2]
```diff
@@ -62,10 +63,18 @@ export class SignalManager {
...
     ({ events: this.events, logger: this.logger } = options);
 
+    // Use a unique subscriber ID for each signals instance, in order to fan-out
+    // all events to each signals instance. This ensures that events always
+    // reach users in a scaled deployment.
+    const id = `signals-${crypto.randomBytes(8).toString('hex')}`;
```
Comment (drodil) [inline plugins/signals-backend/src/service/SignalManager.ts:72]: Hope this is random enough to not collide

Comment (Rugvip) [reply plugins/signals-backend/src/service/SignalManager.ts:72]: 1/18,446,744,073,709,551,615 + birthday paradox, but we won't have too many in flight at once. I think we should be fine x)

File: plugins/signals-backend/src/service/SignalManager.ts (hunk 1 above)
Comment (Rugvip) [reply plugins/signals-backend/src/service/SignalManager.ts:66]: ah, thought I removed that one

**Pull request merged**